File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    record = []
    answer = []
    for i in score:
        if len(record) < k:
            record.append(i)
        else:
            if record[-1] < i:
                record.pop()
                record.append(i)
        record.sort(reverse=True)
        answer.append(record[-1])

    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.ArrayList;
import java.util.Collections;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        ArrayList<Integer> list = new ArrayList<Integer>();
        for (int i = 0; i<score.length; i++){
            list.add(score[i]);
            Collections.sort(list);     
            if(list.size()>k){
                list.remove(0);
            }
            answer[i] = list.get(0);
        }
        return answer;
    }
}